api/migrations/20240906151116-fix_youngs_status_equivalence.js

```diff
@@ -0,0 +1,30 @@
+const { logger } = require("../src/logger");
+const { YoungModel, MissionEquivalenceModel } = require("../src/models");
+
+module.exports = {
+  async up(db, client) {
+    const youngs = await YoungModel.find({
+      statusPhase1: { $in: ["DONE", "EXEMPTED"] },
+      status_equivalence: { $in: ["WAITING_VERIFICATION", "WAITING_CORRECTION"] },
+    });
+
+    let modifiedYoungs = [];
+
+    for (const young of youngs) {
+      const validatedEquivalence = await MissionEquivalenceModel.findOne({
+        youngId: young._id,
+        status: "VALIDATED",
+      });
+
+      if (validatedEquivalence) {
+        young.set("status_equivalence", "VALIDATED");
+        await young.save();
+        modifiedYoungs.push(young);
+      }
+    }
```

Comment:
Quelque chose comme ça ? (pas testé)
```suggestion
 const youngs = await YoungModel.find({
  statusPhase1: { $in: ["DONE", "EXEMPTED"] },
  status_equivalence: { $in: ["WAITING_VERIFICATION", "WAITING_CORRECTION"] },
 }).select({"_id": 1});
 const youngIds = youngs.map(({ _id }) => _id);
 const validatedEquivalences = await MissionEquivalenceModel.find({
    youngId: { $in: youngIds },
    status: "VALIDATED",
  });
  const validatedYoungIds = validatedEquivalences.map(({ youngId }) => youngId)
  await MissionEquivalenceModel.updateMany({
     youngId: { $in: youngIds },
   },
   {
     $set: { status_equivalence: "VALIDATED" }
   }
  );
```